client.lua

```diff
@@ -5,19 +5,40 @@ local pointCamCoords = 75
 local pointCamCoords2 = 0
 local cam1Time = 500
 local cam2Time = 1000
-local choosingSpawn = false
+local isChoosingSpawn = false
 local Houses = {}
 local cam = nil
 local cam2 = nil
 
 -- Functions
 
-local function SetDisplay(bool)
-    choosingSpawn = bool
-    SetNuiFocus(bool, bool)
+-- Stops player from moving while choosing spawn
+local function launchDisableControlsThread()
+    CreateThread(function()
+        while isChoosingSpawn do
+            Wait(0)
+            DisableAllControlActions(0)
+        end
+    end)
+end
+
+---Displays the spawn UI and disables controls
+---@param isShowing boolean
+---@return void
+local function SetDisplay(isShowing)
+    -- sets the global variable to the value passed in
+    isChoosingSpawn = isShowing
+
+    -- launches a thread to disable controls if showing the UI
+    if isShowing then launchDisableControlsThread() end
+
+    -- sets the focus to the NUI window
+    SetNuiFocus(isShowing, isShowing)
+
+    -- sends a message to the NUI window to show or hide the UI
```

Comment:
Not sure we need in-line comments. Consider removing.

Reply:
Can we get a third opinion on the matter? If this is not against style guidelines I would say I disagree with removing. 

**Justification**
I like to make code over simplified so future devs of any level can understand what's happening at a glance.

Reply:
I think we need to assume that readers of the code can understand code. Inline comments are generally avoided because they can differ from the code. So if someone is reading the code they will likely ignore the comments as potential lies that are not to be trusted.

Reply:
That's hilarious I literally watched that video this morning.

I hear what you are saying. Making change now